### PR TITLE
♻️ [MRG] Packaging and console script

### PR DESCRIPTION
**Pull request by hayesall**
## Summary

This refactors the `arxiv_latex_cleaner` into a package, and adds a `setup.py` so this may be installed with a console point entry point.

## Details

- Move `arxiv_latex_cleaner.py` into package directory
- Move `arxiv_latex_cleaner_test.py` into package directory
- Add version information in `arxiv_latex_cleaner/_version.py`
- Drop `__main__` block from `arxiv_latex_cleaner`
- Start using `run_arxiv_cleaner` as a public method
- Fix test to reflect public `run_arxiv_cleaner`
- Add argument parsing to `arxiv_latex_cleaner.__main__`
- Add `setup.py` with an entry point to the main module

One of the biggest changes is the addition of the `setup.py` with an entry point that allows this to be used as a command-line program.

```console
python setup.py install
arxiv_latex_cleaner --help
```

### CODE REVIEW

**Comment by jponttuset** [comment]
Hi @hayesall 
Thank you so much for this PR, it much improves usability, very appreciated!
Could you just patch my two last CLs though, this PR reverts some of the changes.
I'll leave a comment where these changes happened.

Thanks again!

File: arxiv_latex_cleaner/arxiv_latex_cleaner.py [hunk 1]
```diff
@@ -99,14 +99,14 @@ def _remove_command(text, command):
   Regex expression used to match balanced parentheses taken from:
   https://stackoverflow.com/questions/546433/regular-expression-to-match-balanced-parentheses/35271017#35271017
   """
-  return re.sub(r'\\' + command + r'{(?:[^}{]+|{(?:[^}{]+|{[^}{]*})*})*}',
+  return re.sub(r'\\' + command + r'\{(?:[^}{]+|\{(?:[^}{]+|\{[^}{]*\})*\})*\}',
```
Comment (jponttuset) [inline arxiv_latex_cleaner/arxiv_latex_cleaner.py]: Please revert this change

File: arxiv_latex_cleaner/arxiv_latex_cleaner.py [hunk 2]
```diff
@@ -99,14 +99,14 @@ def _remove_command(text, command):
...
                 '', text)
 
 
 def _remove_environment(text, environment):
   """Removes '\\begin{environment}*\\end{environment}' from 'text'."""
   return re.sub(
-      r'\\begin{' + environment + r'}[\s\S]*?\\end{' + environment + r'}',
+      r'\\begin\{' + environment + r'\}[\s\S]*?\\end\{' + environment + r'\}',
```
Comment (jponttuset) [inline arxiv_latex_cleaner/arxiv_latex_cleaner.py]: Please revert this change

File: arxiv_latex_cleaner/arxiv_latex_cleaner.py [hunk 3]
```diff
@@ -153,7 +153,7 @@ def _resize_and_copy_figure(filename,
                             destination_folder,
                             dest_size=600,
                             compress_pdf=False):
-  """Resizes and copies the input figure (either JPG, PNG, or PDF)."""
+  """Copies a file while erasing all the LaTeX comments in its content."""
```
Comment (jponttuset) [inline arxiv_latex_cleaner/arxiv_latex_cleaner.py]: Please revert this change

**Comment by hayesall** [comment]
Hey @jponttuset, thank you for catching the `re.sub` and docstring.

I was three commits behind when I started writing, I probably didn't merge them properly after fetching.

File: arxiv_latex_cleaner/__main__.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    help="Compress PDF images using ghostscript (Linux and Mac only).",
+)
+
+PARSER.add_argument(
+    "--commands_to_delete",
+    nargs="+",
+    default=[],
+    required=False,
+    help=(
+        "LaTeX commands that will be deleted. Useful for e.g. user-defined"
```
Comment (hayesall) [inline arxiv_latex_cleaner/__main__.py]: Looks like I missed a space here after "user-defined", I'll fix this.

**Comment by jponttuset** [comment]
Thanks again @hayesall!
